platform-sdk/docs/proposals/001-tss-rosters-design-proposal/proposal.md

```diff
@@ -0,0 +1,290 @@
+- Roster Adoption: The platform will adopt the last submitted Candidate Roster when it is ready, based on the TSS key
+  material and vote accumulation.
+- Roster Replacement: If a new CAB is submitted before the previous one is adopted, the corresponding new Candidate
+  Roster will replace the old one.
+
+### Public API
+
+A new method will be added to the platform API to allow the App submit a Candidate Roster to the platform:
+
+```java
+//in State
+
+void submitCandidateRoster(@NonNull final Roster candidateRoster);
+```
+
+### Configuration
+
+The `config.txt` file will no longer be used for storing the address book in existing networks. It will only be used for
+the genesis of new networks.
+
+### Data storage
+
+All the data related to Active and Candidate Roster, including the key material being accumulated via TSS messages, will
+always be stored in the State.
+There will not be any other separate artifacts stored elsewhere (e.g., directly on disk.).
+The only artifact that will continue to be stored on disk separately from the State is the `config.txt`, which, as
+explained in *Bootstrapping Genesis for a brand-new network* section below, will only ever be used once during a genesis
+of a brand-new network. The `config.txt` file, or at least its part that describes the address book components, will
+never be used in the life cycle of an existing network after that.
+
+### Startup procedure (pseudo-code)
+
+The pseudocode for the startup procedure will look as follows:
+
+```java
+if(a State
+exists on
+disk){
+
+loadStateFromDisk();
+}else{
+roster =
+
+loadFromConfigTxt();
+
+// Intall the roster as both Active Roster and Candidate Roster in the new state:
+createEmptyState(roster);
+// Note that this Active Roster is missing TSS key material.
+// So the node will only be able to sign using its RSA key, w/o any TSS.
+isGenesis =true;
+        }
+
+        if(
+the Statehas
+Candidate Roster){
+        // Check if this is a software upgrade, and if the TSS protocol
+        // has been launched already, then also check
+        // if Candidate Roster has enough TSS key material/signatures.
+        // Note that we switch to Candidate Roster during a software upgrade unconditionally
+        // until the TSS protocol is actually launched.
+        if(isSoftwareUpgrade /* && Candidate Roster is complete */){
+
+// This MUST be performed under `isSoftwareUpgrade` to ensure that
+// the entire network is being restarted, and so every node adopts the Candidate Roster,
+// and hence no ISSes happen.
+// Modify the state and put Candidate Roster into Active Roster, effectively clearing the Candidate Roster.
+makeCRtheAR();
+// May make a record of the previous Active Roster if necessary (e.g. for PCES replay)
+// Note that the previous Active Roster may lack TSS key material
+// if this is genesis, or if this is a TSS upgrade of an existing network.
+    }
+
+	/*
+	// This block will be uncommented once the TSS protocol is implemented
```

Comment:
I think talking about the conditions relevant in the TSS-Ledger-Id proposal  is confusing and not part of the scope of this proposal.   I'm not going to comment on it any more, but it applies to the rest of the comments in the pseudo code.

Reply:
@anthony-swirldslabs, ok if we just remove all references to `Key Material` from this proposal? Seems to be a common source of confusion.

Reply:
I've removed all references to `Key Material`, to be included in `TSS Ledger ID` proposal if required.

Reply:
Yes, that sounds good to me. The references exist because the Roster design issue explicitly requested answers to questions related to the Roster life-cycle in TSS world. But I'm good if we move those references elsewhere.